go/src/lib.rs

```diff
@@ -428,6 +428,39 @@ pub unsafe extern "C" fn command(
                 command_response.array_elements_len = vec_elements_len_ptr;
                 Ok(Some(command_response))
             }
+            Value::Map(map) => {
+                let mut vec_elements_len: Vec<c_long> = Vec::<c_long>::with_capacity(map.len() * 2);
+
+                let result = map
+                    .into_iter()
+                    .flat_map(|(key, value)| {
+                        let key = <Vec<u8>>::from_redis_value(&key)
```

Comment:
This seems like reducing a redis::Value to always a string.
That may serve a specific purpose now of Map<String, String>, but I dont think that would be a right implementation.

We should try to convert the redis::Value to CommandResponse so that the generic nature is maintained.

We are working towards that as you can check in the PR: https://github.com/valkey-io/valkey-glide/pull/2289